File: backend/src/main/java/com/evomind/entity/CognitiveConflict.java
package com.evomind.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

/**
 * 认知冲突实体
 * 记录新卡片与用户认知画像的冲突
 */
@Entity
@Table(name = "cognitive_conflicts")
@Getter
@Setter
public class CognitiveConflict extends BaseEntity {

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "card_id", nullable = false)
    private Long cardId;

    @Column(name = "profile_id", nullable = false)
    private Long profileId;

    @Column(name = "conflict_type", length = 50)
    private String conflictType;

    @Column(name = "conflict_description", length = 500)
    private String conflictDescription;

    @Column(name = "user_belief", length = 1000)
    private String userBelief;

    @Column(name = "card_viewpoint", length = 1000)
    private String cardViewpoint;

    @Column(name = "conflict_score", precision = 5, scale = 4)
    private BigDecimal conflictScore;

    @Column(name = "is_acknowledged")
    private Boolean isAcknowledged = false;

    @Column(name = "is_dismissed")
    private Boolean isDismissed = false;

    @Column(name = "ai_analysis", length = 2000)
    private String aiAnalysis;

    @Column(name = "topic", length = 100)
    private String topic;
}

File: android-app/app/src/main/java/com/evomind/domain/model/CognitiveConflict.kt
package com.evomind.domain.model

import java.math.BigDecimal
import java.time.LocalDateTime

/**
 * 认知冲突数据模型（新卡片与用户认知体系的冲突）
 */
data class CognitiveConflict(
    val id: Long,
    val cardId: Long,
    val cardTitle: String,
    val cardViewpoint: String,
    val profileId: Long,
    val topic: String,
    val userBelief: String,
    val conflictType: ConflictType,
    val conflictDescription: String,
    val conflictScore: BigDecimal,
    val isAcknowledged: Boolean,
    val isDismissed: Boolean,
    val aiAnalysis: String,
    val createdAt: LocalDateTime
) {
    /**
     * 获取冲突严重程度
     */
    fun getSeverity(): ConflictSeverity {
        return when {
            conflictScore.toDouble() >= 0.8 -> ConflictSeverity.HIGH
            conflictScore.toDouble() >= 0.6 -> ConflictSeverity.MEDIUM
            else -> ConflictSeverity.LOW
        }
    }

    /**
     * 获取冲突类型描述
     */
    fun getConflictTypeDescription(): String {
        return conflictType.description
    }
}

/**
 * 用户认知画像
 */
data class UserCognitiveProfile(
    val id: Long,
    val topic: String,
    val coreBelief: String,
    val beliefType: BeliefType,
    val confidenceLevel: BigDecimal,
    val evidenceCount: Int
)

/**
 * 信念类型
 */
enum class BeliefType(val description: String) {
    STRONG_CONViction("强信念"),
    MODERATE_STANCE("中等立场"),
    EXPLORING("探索中"),
    TENTATIVE("暂定")
}

/**
 * 冲突严重程度
 */
enum class ConflictSeverity(val level: Int, val description: String) {
    HIGH(3, "高"),
    MEDIUM(2, "中"),
    LOW(1, "低")
}
